shared/util/load-perf.js

```diff
@@ -40,5 +41,53 @@ export default function () {
   }
 
   window.KBPERF = onPerf
+  window.shallowEqualDebug = shallowEqualDebug
```

Comment:
add a helper so we can debug when shallowEqual is failing. it'll spit out the names of the keys and values so you can see whats going on